### PR TITLE
implement initial script

### PR DESCRIPTION
**Pull request by Godley**

### CODE REVIEW

File: pod.yaml
Comment (stuart-warren) [inline pod.yaml]: do we need to specify `kube-system` here?

File: pod.yaml
Comment (stuart-warren) [inline pod.yaml]: This is not good to use for opensource software, this wont work for anyone else

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: We should probably set a default but allow overriding the namespace that kubectl uses in this script

**Review by stuart-warren — CHANGES_REQUESTED** [review]

File: pod.yaml
Comment (Godley) [reply pod.yaml]: guess not? would we need to put it in a namespace in the script though?

Comment (stuart-warren) [reply pod.yaml]: probably

**Comment by Godley** [comment]
@stuart-warren PTAL

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: Need to be able to override this

```
SCHEDULER_NAMESPACE="${SCHEDULER_NAMESPACE:-kube-system}"
```

File: pod.yaml
Comment (stuart-warren) [inline pod.yaml]: not sure where you got this default from, 3.0 is the latest https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/pause

need to be able to override this somehow though. maybe some sed or envsubst magic in the script

**Review by stuart-warren — CHANGES_REQUESTED** [review]

File: pod.yaml
Comment (Godley) [reply pod.yaml]: ta, think it was a random old issue + wasn't sure how you navigate gcr so didn't delve further into latest tag.
Do we need to override the tag or the image?

Comment (stuart-warren) [reply pod.yaml]: I'd guess the whole image string including the tag

File: Dockerfile
Comment (stuart-warren) [inline Dockerfile]: Should this be
```
COPY pod.yaml /
```
rather than copying to a relative path?

File: pod.yaml
Comment (mikebryant) [inline pod.yaml]: This isn't going to work in our environments without Internet.

We'll need to either hardcode this to our mirror names, or make it a parameter

File: kube-scheduler-healthcheck
Comment (mikebryant) [inline kube-scheduler-healthcheck]: Use kubectl jsonpath?

**Review by mikebryant — CHANGES_REQUESTED** [review]

File: pod.yaml
Comment (Godley) [reply pod.yaml]: What have we done elsewhere/which would you recommend? I had it hard coded but Stuart pointed out that won't work in open source.

Comment (Godley) [reply pod.yaml]: wondering if it'd be better to merge this into the script rather than having a separate asset

Comment (mikebryant) [reply pod.yaml]: Mhm, that should be fairly easy.
Just write it out to `/tmp` using heredocs - environment variable interpolation will just work

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: Still need to sed/envsubst the image name into pod.yaml

**Review by stuart-warren — CHANGES_REQUESTED** [review]

File: pod.yaml
Comment (Godley) [reply pod.yaml]: any example of that? googling heredocs mostly just comes up with a wikipedia entry, not sure if that's a bash tool or

Comment (Godley) [reply pod.yaml]: nvm, think I got it

File: kube-scheduler-healthcheck
Comment (mikebryant) [inline kube-scheduler-healthcheck]: Move `google_containers` to the next line

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: I don't think this needs to be split into 2 parts

File: kube-scheduler-healthcheck
Comment (mikebryant) [inline kube-scheduler-healthcheck]: `-o jsonpath={.status.phase}`

File: kube-scheduler-healthcheck
Comment (Godley) [reply kube-scheduler-healthcheck]: @stuart-warren recommended getting rid of the split all together. Which way are we going?

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: should probably add some form of request timeout to all kubectl commands like https://gitlab.tech.lastmile.com/kubernetes/openstack-deploy/merge_requests/70

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: declare an alias or variable to reduce the duplication

Comment (Godley) [reply kube-scheduler-healthcheck]: for the whole line?

Comment (stuart-warren) [reply kube-scheduler-healthcheck]: for `kubectl --namespace $TEST_POD_NAMESPACE --request-timeout=30s`

**Comment by Godley** [comment]
@mikebryant @stuart-warren PTAL

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: sorry, it seems I was unclear.

The alias was for all instances of kubectl as they all use those same 2 flags, not just the delete pod one

`alias kubectl='kubectl --namespace $TEST_POD_NAMESPACE --request-timeout=30s'`

Comment (Godley) [reply kube-scheduler-healthcheck]: aaaah ok, sorry I skim read the command in your previous comment

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: sorry, this actually needs to be:
```
alias kubectl="kubectl --namespace \$TEST_POD_NAMESPACE --request-timeout=30s"
```
single quotes won't work and need to escape the dollar

Please use long form parameters in scripts to make as clear as possible when reading

Comment (stuart-warren) [reply kube-scheduler-healthcheck]: and just make the alias `k` or `kubectl`, no need to add other stuff to it

Comment (Godley) [reply kube-scheduler-healthcheck]: why? to me `kubectl_and_args` is clearer than just `kubectl` or `k`. could miss that line and assume it's default `kubectl`.

**Review by mikebryant — APPROVED** [review]

**Comment by mikebryant** [comment]
Squash before merge?

**Comment by Godley** [comment]
Squashed, just going to try it out locally to check syntax errors etc.

File: kube-scheduler-healthcheck
Comment (stuart-warren) [inline kube-scheduler-healthcheck]: should be `3.0` not `3.0.0`

**Comment by Godley** [comment]
@stuart-warren we done yet? 😛

**Comment by stuart-warren** [comment]
I hope so! LGTM

**Comment by Godley** [comment]
Think you have to press approve? @stuart-warren I don't have write access so presumably you or @mikebryant has to merge it

**Pull request merged**
